pkg/identity/identity_test.go

```diff
@@ -87,50 +92,129 @@ func TestFullIdentityFromPEM(t *testing.T) {
 	assert.Equal(t, leafKey, fullIdent.Key)
 }
 
-func TestConfig_SaveIdentity(t *testing.T) {
+func TestConfig_Save_with_extension(t *testing.T) {
 	ctx := testcontext.New(t)
 	defer ctx.Cleanup()
 
-	ic := &identity.Config{
-		CertPath: ctx.File("chain.pem"),
-		KeyPath:  ctx.File("key.pem"),
-	}
-	fi := pregeneratedIdentity(t)
+	testidentity.CompleteIdentityVersionsTest(t, func(t *testing.T, version storj.IDVersion, ident *identity.FullIdentity) {
+		identCfg := &identity.Config{
+			CertPath: ctx.File("chain.pem"),
+			KeyPath:  ctx.File("key.pem"),
+		}
 
-	chainPEM := bytes.NewBuffer([]byte{})
-	assert.NoError(t, pkcrypto.WriteCertPEM(chainPEM, fi.Leaf))
-	assert.NoError(t, pkcrypto.WriteCertPEM(chainPEM, fi.CA))
+		{ // pre-save version assertions
+			assert.Equal(t, version.Number, ident.ID.Version().Number)
 
-	privateKey := fi.Key
-	assert.NotEmpty(t, privateKey)
+			caVersion, err := storj.IDVersionFromCert(ident.CA)
+			require.NoError(t, err)
+			assert.Equal(t, version.Number, caVersion.Number)
 
-	keyPEM := bytes.NewBuffer([]byte{})
-	assert.NoError(t, pkcrypto.WritePrivateKeyPEM(keyPEM, privateKey))
+			versionExt := tlsopts.NewExtensionsMap(ident.CA)[extensions.IdentityVersionExtID.String()]
+			if ident.ID.Version().Number == 1 {
+				require.NotEmpty(t, versionExt)
+				assert.Equal(t, ident.ID.Version().Number, storj.IDVersionNumber(versionExt.Value[0]))
+			} else {
+				assert.Empty(t, versionExt)
+			}
+		}
 
-	{ // test saving
-		err := ic.Save(fi)
-		assert.NoError(t, err)
+		{ // test saving
+			err := identCfg.Save(ident)
+			assert.NoError(t, err)
 
-		certInfo, err := os.Stat(ic.CertPath)
-		assert.NoError(t, err)
+			certInfo, err := os.Stat(identCfg.CertPath)
+			assert.NoError(t, err)
```

Comment:
we should probably use [require.NoError](https://godoc.org/github.com/stretchr/testify/require#NoError) instead of assert.NoError

Reply:
:+1:

Reply:
I updated several other places where we should be using `require` instead of `assert`. I think using `require` here might not be necessary, but I could be wrong. There are two tests here that are sharing a `*testing.T`.